src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -2,23 +2,207 @@
+                        Type = new CSRef(ptr.Constant || baseType.Constant ? CSRef.Type.In : CSRef.Type.Ref,
+                            vector),
+                        Length = null
+                    };
+                    spanParameters[i] = parameter with { Type = new CSSpan(vector, ptr.Constant || baseType.Constant), Length = null };
+                    arrayParameters[i] = parameter with { Type = new CSArray(vector, ptr.Constant || baseType.Constant), Length = null };
+                    nameTable.Rename(parameter, parameter.Name + "_ptr");
+                    overloadedParams.Add((parameter, singleParameters[i], firstElem));
+                }
+            }
+
+            if (overloadedParams.Count == 0)
+            {
+                newOverloads = null;
+                return false;
+            }
+
+            string overloadName = overload.OverloadName.Remove(m.Index + 2, 1);
+            newOverloads = new List<Overload>()
+            {
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = singleParameters.Except(countParameters).ToArray(),
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SingleVectorLayer(countParameters, overloadedParams)
+                },
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = spanParameters,
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SpanArrayVectorLayer(countParameters, overloadedParams)
+                },
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = arrayParameters,
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SpanArrayVectorLayer(countParameters, overloadedParams)
+                },
+            };
+            return true;
+        }
+
+        public record SingleVectorLayer(
+            List<Parameter> CountParameters,
+            List<(Parameter ptrParam, Parameter vectorParam, string firstElem)> OverloadedParameters) : IOverloadLayer
+        {
+            private CsScope _csScope;
+            public void WritePrologue(IndentedTextWriter writer, NameTable nameTable)
+            {
+                foreach (var refParam in CountParameters)
+                {
+                    writer.WriteLine($"{refParam.Type.ToCSString()} {nameTable[refParam]} = 1;");
+                }
+                foreach (var (ptrParam, vectorParam, firstElem) in OverloadedParameters)
+                {
+                    writer.WriteLine($"fixed ({ptrParam.Type.ToCSString()} {nameTable[ptrParam]} = &{nameTable[vectorParam]}{firstElem})");
```

Comment:
We don't really need the `FirstElem` parameter. We could alternatively do it like:
```cs
fixed(TVector* vector_ptr = &vector)
{
    TElement* ptr = (TElement*)vector_ptr;
    // The rest of the overload code...
}
```
Not sure this is better, it's just an alternative to consider.

Reply:
The problem with this is if we want to support multiple vector parameters for one method (which im not even sure we want to) We need an extra for loop for this. We also need to set up a bunch of new parameters for this to work. Unless we are ok with the nametable not including variables that are only to be used within a single overloader.